src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -759,21 +768,146 @@ public boolean isKnownDatasource( UUID studyId, String participantId, String dat
                 && participantDevices.get( participantId ).containsKey( datasourceId );
     }
 
+    private void deleteStudyData(
+            UUID studyId,
+            java.util.Optional<String> participantId,
+            com.openlattice.data.DeleteType deleteType,
+            String userToken ) {
+        try {
+            // load api for actions authenticated by the user
+            ApiClient userApiClient = new ApiClient( () -> userToken
+            );
+            SearchApi userSearchApi = userApiClient.getSearchApi();
+            EntitySetsApi userEntitySetsApi = userApiClient.getEntitySetsApi();
+            DataApi userDataApi = userApiClient.getDataApi();
+
+            // load api for actions authenticated by chronicle
+            // because of the way we do things right now, only the chronicle
+            // user can have permissions to delete from the entity set.
+            // To be replaced fully with user authentication after refactoring.
+            ApiClient chronicleApiClient = apiClientCache.get( ApiClient.class );
+            DataApi chronicleDataApi = chronicleApiClient.getDataApi();
+
+            String participantsEntitySetName = getParticipantEntitySetName( studyId );
+            UUID participantsEntitySetId = userEntitySetsApi.getEntitySetId( participantsEntitySetName );
+            if ( participantsEntitySetId == null ) {
+                logger.error( "unable to get the participants EntitySet id for studyId = {}", studyId );
+            }
+
+            // get a set of all participants to remove:
+            Set<UUID> participantsToRemove = new HashSet<>();
+            if ( participantId.isPresent() ) {
+                // if participantId: add to set
+                UUID participantEntityKeyId = getParticipantEntityKeyId( participantId.get(), studyId );
+                if (participantEntityKeyId == null) {
+                    logger.error("unable to delete participant {}: participant does not exist.", participantId);
+                }
+                participantsToRemove.add( participantEntityKeyId );
+            } else {
+                // if no participant Id: load all participants and add to set
+                chronicleDataApi
```

Comment:
(just a spacing thing) -- I think this is a little easier to read with nesting layers somewhat respected, so formatted like this
```
        chronicleDataApi
                .loadEntitySetData( participantsEntitySetId, FileType.json, userToken )
                .forEach( entity -> entity.get( ID_FQN ).forEach( personId -> 
                        participantsToRemove.add( UUID.fromString( personId.toString() ) )
                )
        );
```